replica/database.cc

```diff
@@ -1144,7 +1144,8 @@ std::vector<sstring> database::get_non_local_strategy_keyspaces() const {
     std::vector<sstring> res;
     res.reserve(_keyspaces.size());
     for (auto const& i : _keyspaces) {
-        if (i.second.get_replication_strategy().get_type() != locator::replication_strategy_type::local) {
+        auto&& rs = i.second.get_replication_strategy();
+        if (rs.get_type() != locator::replication_strategy_type::local && !rs.is_per_table()) {
```

Comment:
maybe the function should be renamed too, since `non_local_strategy` technically includes per-table strategies too that are non-local.
e.g. `get_non_local_token_based_strategy_keyspaces`

Reply:
Agreed